### PR TITLE
Refactor app tokens

### PR DESCRIPTION
**Pull request by IKarbowiak**
Store app tokens hashes instead of plain text.

# Impact

* [ ] New migrations
* [ ] New/Updated API fields or mutations
* [ ] Deprecated API fields or mutations
* [ ] Removed API types, fields, or mutations
* [ ] Documentation needs to be updated

# Pull Request Checklist

* [ ] Privileged queries and mutations are guarded by proper permission checks
* [ ] Database queries are optimized and the number of queries is constant
* [ ] Database migration files are up to date
* [ ] The changes are tested
* [ ] GraphQL schema and type definitions are up to date
* [ ] Changes are mentioned in the changelog

### CODE REVIEW

**Comment by patrys** [comment]
Task linked: [SALEOR-6362 Storing app tokens in plain text](https://app.clickup.com/t/2549495/SALEOR-6362)

File: saleor/graphql/context.py [hunk 1]
```diff
@@ -16,18 +17,23 @@ def get_context_value(request):
...
 
-def get_app(auth_token) -> Optional[App]:
-    tokens = AppToken.objects.filter(auth_token=auth_token).values("pk")
+def get_app(raw_auth_token) -> Optional[App]:
+    tokens = AppToken.objects.filter(token_last_4=raw_auth_token[-4:]).values_list(
+        "id", "auth_token"
+    )
+    token_ids = [
+        id for id, auth_token in tokens if check_password(raw_auth_token, auth_token)
+    ]
```
Comment (IKarbowiak) [inline saleor/graphql/context.py:26]: I need to add iteration, as we right now do not have the raw auth token values saved, only the last 4 chars. IMO it shouldn't cause any performance problems as we iterate only through objects, in which the last 4 chars match the last chars of a given token.
But if anyone has a better idea, I'm open for any suggestions.

File: saleor/app/management/commands/create_app.py [hunk 2]
```diff
@@ -59,9 +59,9 @@ def handle(self, *args: Any, **options: Any) -> Optional[str]:
         permissions = clean_permissions(permissions)
         app = App.objects.create(name=name, is_active=is_active)
         app.permissions.set(permissions)
-        token_obj = app.tokens.create()
+        _token_obj, auth_token = app.tokens.create()
```
Comment (L3str4nge) [inline saleor/app/management/commands/create_app.py:62]:
```suggestion
        _,  auth_token = app.tokens.create()
```
We don't use `_token_obj`

Comment (L3str4nge) [inline saleor/app/management/commands/create_app.py:62]:
```suggestion
        _, auth_token = app.tokens.create()
```
We don't use `_token_obj` variable

File: saleor/app/management/commands/install_app.py [hunk 3]
```diff
@@ -53,11 +53,10 @@ def handle(self, *args: Any, **options: Any) -> Optional[str]:
             app_job.permissions.set(permissions)
 
         try:
-            app = install_app(app_job, activate)
+            app, token = install_app(app_job, activate)
```
Comment (L3str4nge) [inline saleor/app/management/commands/install_app.py:56]:
```suggestion
            _, token = install_app(app_job, activate)
```

File: saleor/graphql/context.py (hunk 1 above)
Comment (fowczarek) [reply saleor/graphql/context.py:26]: We should be sure that this code is not executed before all requests from non-app users.

File: saleor/graphql/context.py [hunk 4]
```diff
@@ -16,18 +17,23 @@ def get_context_value(request):
...
+        Exists(tokens.filter(id__in=token_ids, app_id=OuterRef("pk"))), is_active=True
     ).first()
 
 
 def set_app_on_context(request):
     if request.path == API_PATH and not hasattr(request, "app"):
         request.app = None
         auth_token = get_token_from_request(request)
-        if auth_token and len(auth_token) == 30:
+        if auth_token and len(auth_token) <= AppToken.auth_token.field.max_length:
```
Comment (fowczarek) [inline saleor/graphql/context.py:36]: @korycins Could you confirm that the change in condition is safe?
As far as I remember token shorter than 30 char is a token from external authorization.  

@IKarbowiak Did you test it with external auth?

Comment (IKarbowiak) [reply saleor/graphql/context.py]: I reverse this change. It was not needed.

**Review by korycins — APPROVED** [review]

**Review by Qasem-h — APPROVED** [review]

**Review by fowczarek — APPROVED** [review]

**Review by maarcingebala — APPROVED** [review]

**Review by L3str4nge — APPROVED** [review]

**Pull request merged**